<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Bruce E. Kendall" />

<meta name="date" content="2021-06-27" />

<title>Rationale and Roadmap</title>

<script src="data:application/javascript;base64,Ly8gUGFuZG9jIDIuOSBhZGRzIGF0dHJpYnV0ZXMgb24gYm90aCBoZWFkZXIgYW5kIGRpdi4gV2UgcmVtb3ZlIHRoZSBmb3JtZXIgKHRvCi8vIGJlIGNvbXBhdGlibGUgd2l0aCB0aGUgYmVoYXZpb3Igb2YgUGFuZG9jIDwgMi44KS4KZG9jdW1lbnQuYWRkRXZlbnRMaXN0ZW5lcignRE9NQ29udGVudExvYWRlZCcsIGZ1bmN0aW9uKGUpIHsKICB2YXIgaHMgPSBkb2N1bWVudC5xdWVyeVNlbGVjdG9yQWxsKCJkaXYuc2VjdGlvbltjbGFzcyo9J2xldmVsJ10gPiA6Zmlyc3QtY2hpbGQiKTsKICB2YXIgaSwgaCwgYTsKICBmb3IgKGkgPSAwOyBpIDwgaHMubGVuZ3RoOyBpKyspIHsKICAgIGggPSBoc1tpXTsKICAgIGlmICghL15oWzEtNl0kL2kudGVzdChoLnRhZ05hbWUpKSBjb250aW51ZTsgIC8vIGl0IHNob3VsZCBiZSBhIGhlYWRlciBoMS1oNgogICAgYSA9IGguYXR0cmlidXRlczsKICAgIHdoaWxlIChhLmxlbmd0aCA+IDApIGgucmVtb3ZlQXR0cmlidXRlKGFbMF0ubmFtZSk7CiAgfQp9KTsK"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>





<style type="text/css">
/* for pandoc --citeproc since 2.11 */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}
</style>

<link rel="stylesheet" href="data:text/css,body%20%7B%0Abackground%2Dcolor%3A%20%23fff%3B%0Amargin%3A%201em%20auto%3B%0Amax%2Dwidth%3A%20700px%3B%0Aoverflow%3A%20visible%3B%0Apadding%2Dleft%3A%202em%3B%0Apadding%2Dright%3A%202em%3B%0Afont%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0Afont%2Dsize%3A%2014px%3B%0Aline%2Dheight%3A%201%2E35%3B%0A%7D%0A%23TOC%20%7B%0Aclear%3A%20both%3B%0Amargin%3A%200%200%2010px%2010px%3B%0Apadding%3A%204px%3B%0Awidth%3A%20400px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Aborder%2Dradius%3A%205px%3B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Afont%2Dsize%3A%2013px%3B%0Aline%2Dheight%3A%201%2E3%3B%0A%7D%0A%23TOC%20%2Etoctitle%20%7B%0Afont%2Dweight%3A%20bold%3B%0Afont%2Dsize%3A%2015px%3B%0Amargin%2Dleft%3A%205px%3B%0A%7D%0A%23TOC%20ul%20%7B%0Apadding%2Dleft%3A%2040px%3B%0Amargin%2Dleft%3A%20%2D1%2E5em%3B%0Amargin%2Dtop%3A%205px%3B%0Amargin%2Dbottom%3A%205px%3B%0A%7D%0A%23TOC%20ul%20ul%20%7B%0Amargin%2Dleft%3A%20%2D2em%3B%0A%7D%0A%23TOC%20li%20%7B%0Aline%2Dheight%3A%2016px%3B%0A%7D%0Atable%20%7B%0Amargin%3A%201em%20auto%3B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dcolor%3A%20%23DDDDDD%3B%0Aborder%2Dstyle%3A%20outset%3B%0Aborder%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0Aborder%2Dwidth%3A%202px%3B%0Apadding%3A%205px%3B%0Aborder%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0Aborder%2Dwidth%3A%201px%3B%0Aborder%2Dstyle%3A%20inset%3B%0Aline%2Dheight%3A%2018px%3B%0Apadding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0Aborder%2Dleft%2Dstyle%3A%20none%3B%0Aborder%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Ap%20%7B%0Amargin%3A%200%2E5em%200%3B%0A%7D%0Ablockquote%20%7B%0Abackground%2Dcolor%3A%20%23f6f6f6%3B%0Apadding%3A%200%2E25em%200%2E75em%3B%0A%7D%0Ahr%20%7B%0Aborder%2Dstyle%3A%20solid%3B%0Aborder%3A%20none%3B%0Aborder%2Dtop%3A%201px%20solid%20%23777%3B%0Amargin%3A%2028px%200%3B%0A%7D%0Adl%20%7B%0Amargin%2Dleft%3A%200%3B%0A%7D%0Adl%20dd%20%7B%0Amargin%2Dbottom%3A%2013px%3B%0Amargin%2Dleft%3A%2013px%3B%0A%7D%0Adl%20dt%20%7B%0Afont%2Dweight%3A%20bold%3B%0A%7D%0Aul%20%7B%0Amargin%2Dtop%3A%200%3B%0A%7D%0Aul%20li%20%7B%0Alist%2Dstyle%3A%20circle%20outside%3B%0A%7D%0Aul%20ul%20%7B%0Amargin%2Dbottom%3A%200%3B%0A%7D%0Apre%2C%20code%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0Aborder%2Dradius%3A%203px%3B%0Acolor%3A%20%23333%3B%0Awhite%2Dspace%3A%20pre%2Dwrap%3B%20%0A%7D%0Apre%20%7B%0Aborder%2Dradius%3A%203px%3B%0Amargin%3A%205px%200px%2010px%200px%3B%0Apadding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0Abackground%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0Acode%20%7B%0Afont%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0Afont%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0Apadding%3A%202px%200px%3B%0A%7D%0Adiv%2Efigure%20%7B%0Atext%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0Abackground%2Dcolor%3A%20%23FFFFFF%3B%0Apadding%3A%202px%3B%0Aborder%3A%201px%20solid%20%23DDDDDD%3B%0Aborder%2Dradius%3A%203px%3B%0Aborder%3A%201px%20solid%20%23CCCCCC%3B%0Amargin%3A%200%205px%3B%0A%7D%0Ah1%20%7B%0Amargin%2Dtop%3A%200%3B%0Afont%2Dsize%3A%2035px%3B%0Aline%2Dheight%3A%2040px%3B%0A%7D%0Ah2%20%7B%0Aborder%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Apadding%2Dbottom%3A%202px%3B%0Afont%2Dsize%3A%20145%25%3B%0A%7D%0Ah3%20%7B%0Aborder%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0Apadding%2Dtop%3A%2010px%3B%0Afont%2Dsize%3A%20120%25%3B%0A%7D%0Ah4%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0Amargin%2Dleft%3A%208px%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Ah5%2C%20h6%20%7B%0Aborder%2Dbottom%3A%201px%20solid%20%23ccc%3B%0Afont%2Dsize%3A%20105%25%3B%0A%7D%0Aa%20%7B%0Acolor%3A%20%230033dd%3B%0Atext%2Ddecoration%3A%20none%3B%0A%7D%0Aa%3Ahover%20%7B%0Acolor%3A%20%236666ff%3B%20%7D%0Aa%3Avisited%20%7B%0Acolor%3A%20%23800080%3B%20%7D%0Aa%3Avisited%3Ahover%20%7B%0Acolor%3A%20%23BB00BB%3B%20%7D%0Aa%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0Aa%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0Atext%2Ddecoration%3A%20underline%3B%20%7D%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%0Acode%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%0A" type="text/css" />




</head>

<body>




<h1 class="title toc-ignore">Rationale and Roadmap</h1>
<h4 class="author">Bruce E. Kendall</h4>
<h4 class="date">2021-06-27</h4>



<p>Why do we need a new package for “Building and Analyzing Matrix Population Models?” After all, R is built on matrix manipulation, and a number of packages exist that support ecological population modeling, including matrix population models (see next section).</p>
<p>However, a matrix <em>population</em> model is not just a matrix: it synthesizes information about age-, stage-, and/or size-dependent demography, development (growth), and movement; may be time-varying, and may contain rates that are density- or frequency-dependent. The existing tools for constructing and analyzing such models either require that the user do this synthesizing themselves or put strong constraints on the structure of the model. In a recent review of the literature on published matrix population models (MPMs), many constructed by non-modelers (i.e., the field biologists who studied the population), I and my co-authors found that a large fraction of MPMs were constructed incorrectly <span class="citation">(Kendall et al. 2019)</span>. This is probably due in part to the fact that most textbook descriptions of MPM construction are incomplete, inaccessible to non-modelers, and/or incoherent. Furthermore, some of the steps involved in converting biological parameters into matrix elements require actions that are, at first encounter, biologically counter-intuitive. These sticking points are not deep biological insights, but rather just accounting issues. Thus, the first objective of <strong>mpmtools</strong> is to help biologists construct MPMs from underlying estimates of demographic and development rates without needing to engage with the troubling accounting issues.</p>
<p>In addition, the standard ways of setting up MPMs is problematic in many cases. For example, in a long-lived animal species, a fully age-structured MPM with many dozens of age classes is cumbersome, prone to data-entry errors, and difficult to extract meaningful biological insight from. Thus, many biologists gravitate towards stage-structured models with a small number of biologically meaningful stage classes (juvenile, adult, etc.). While it is possible (although not trivial) to construct a stage-structured model that matches the <em>asymptotic</em> properties of the age-structured population being approximated, the stage-structured model will do a poor job of approximating other quantities of interest, such as transient dynamics or life-history parameters. A powerful solution is to construct MPMs that account for <em>both</em> age and stage (especially if there is heterogeneity in development rates); but this requires even more attention to accounting details than the standard models, and requires active programming skills to construct and analyze. Thus, the second objective of <strong>mpmtools</strong> is to help biologists construct and analyze MPMs that simultaneously account for, and report results in relation to, broad stage classes in addition to age, or broad size classes in addition to continuous size.</p>
<p>Finally, there are a vast number of biologically useful analyses one can do with an MPM (e.g., see Caswell 2001 and his many subsequent publications). However, most MPM analyses of actual populations (as opposed to methods or theory papers) only look at the asymptotic growth rate (<span class="math inline">\(\lambda\)</span>) and its sensitivity to parameters, or in some cases at stochastic simulations to estimate extinction risk. Implementing the “nonstandard” analyses requires not only a conceptual understanding but also the ability to write (often complex) computer code. Many of these analyses could become standard if the programming hurdle were overcome, just as software implementations of generalized linear models (such as <code>glm()</code>) have allowed ecologists to perform statistical analyses that better match the structure of their data. Thus, the final objective of <strong>mpmtools</strong> is to help biologists learn more about their populations by making advanced analyses more accessible.</p>
<div id="related-packages-and-software" class="section level2">
<h2>Related packages and software</h2>
<dl>
<dt>Base R</dt>
<dd>Base R includes functions for creating matrices (<code>matrix()</code>) and performing eigenvalue/vector analysis (<code>eigen()</code>). It also has various matrix product operators, including the inner product (aka “matrix multiplication;” <code>%*%</code>), the Hadamard product (entry-wise multiplication; <code>*</code>), and the outer product (<code>%o%</code>). These form the basis for the analysis of MPMs, but do not provide any explicitly ecological tools.
</dd>
</dl>
<div id="packages-and-software-for-ecological-population-modeling" class="section level3">
<h3>Packages and software for ecological population modeling</h3>
<dl>
<dt><strong>popbio</strong>: Construction and Analysis of Matrix Population Models</dt>
<dd><strong>popbio</strong> <span class="citation">(Stubben and Milligan 2007)</span> provides an R implementation of the Matlab routines from <span class="citation">Caswell (2001)</span> and <span class="citation">Morris and Doak (2002)</span>. It does not, in general, provide code for other analyses in <span class="citation">Caswell (2001)</span> that did not come with Matlab code. The only extensions are some plotting functions and <code>vitalsens()</code>, a function to calculate the sensitivity and elasticity of <span class="math inline">\(\lambda\)</span> to the underlying demographic parameters. For the latter, the user needs to provide a symbolic representation of the MPM, together with parameter values. The package is actively maintained with bug fixes and updates to keep up with changes in base R and other dependencies, but no new functionality has been added since 2011. The underlying code is variable in the extent to which it makes use of R’s efficiencies, and in the amount of input checking.
</dd>
<dt>popdemo: Demographic Modelling Using Projection Matrices</dt>
<dd><strong>popdemo</strong> <span class="citation">(Stott, Hodgson, and Townley 2018)</span> provides functions for projecting MPMs and analyzing <span class="math inline">\(\lambda\)</span>. In addition, it provides a variety of functions related to transient dynamics and perturbation analysis. The package has routines for testing matrix primitivity etc., and uses an S4 class for population projections that stores the matrices, population vectors, and a number of statistics. The package is under reasonably active development, although the most recent addition is the projection features, which arguably are least innovative (in purpose, if not in implementation).
</dd>
<dt>primer: Functions and data for “A Primer of Ecology with R”</dt>
<dd>This package accompanies a book <span class="citation">(Stevens 2009)</span> that covers a wide range of introductory topics in ecological modeling. MPMs are covered in chapter 2; the one function provided for this chapter calculates all the quantities that can be extracted from the eigenanalysis. Note that the example shown in the book for parameterizing a model from data fails to include a survival in the reproduction transtions.
</dd>
<dt>IPMpack: Construction and analysis of integral projection models</dt>
<dd><strong>IPMpack</strong> <span class="citation">(Metcalf et al. 2014)</span> implements <em>integral projection models</em>, in which an individual’s state is conceptually a continuous variable but is implemented as a high-dimensional MPM. The package is focused on getting from raw data to analysis results (indeed, it is quite challenging to use the package to build a model with predetermined parameter values). It contains an innovative S4 class for doing this. Unfortunately, development has stopped, and it is not clear that there is even active bug support. The package does not support some common types of life histories, such as monocarpic perennials, and has not kept up with the state of the art in model structure and analysis as described in <span class="citation">Ellner, Childs, and Rees (2016)</span> (but note that the latter’s programming is <em>not</em> state of the art).
</dd>
<dt>pop: A Flexible Syntax for Population Dynamic Modelling</dt>
<dd><strong>pop</strong> <span class="citation">(Golding 2016)</span> is an intriguing attempt to create a common interface framework for a wide range of population models, including MPMs. For MPMs, it appears that users still need to fully understand what should go into each transition element, which is one of the challenges we identified in <span class="citation">Kendall et al. (2019)</span>. Furthermore, development appears to have ceased at version 0.1 (which incoporporates simple MPMs and discrete patches of habitat) in 2016.
</dd>
<dt>Rramas: Matrix Population Models</dt>
<dd><strong>Rramas</strong> <span class="citation">(de la Cruz Rot 2017)</span> projects MPMs with environmental and demographic stochasticity. It is meant to reproduce the simulation functionality of the commercial program Ramas, but does not appear to have any connection to that program.
</dd>
<dt>population: Models for Simulating Populations</dt>
<dd><strong>population</strong> <span class="citation">(Chapron 2018)</span> appears to be a generalization of <strong>pop.wolf</strong>. It provides a function to simulate a density-independent age-structured MPM with environmental stochasticity.
</dd>
<dt>PVA: Population viability analysis in R</dt>
<dd><strong>PVA</strong> <span class="citation">(Kendall 2018)</span> contains tools for simulating and analyzing simple population models, with a focus on demonstrations for teaching purposes. It does include functions for working with MPMs, representing my early attempts to address some of the issues that <strong>mpmtools</strong> is aimed at. These will probably be removed once <strong>mpmtools</strong> is mature.
</dd>
<dt>Vortex: Population Viability Analysis software</dt>
<dd>Vortex (<a href="http://www.vortex10.org" class="uri">http://www.vortex10.org</a>) is a free (but not open-source) standalone program to perform population viability analysis on age-structured models. The user only needs to input fertility and survival rates, and doesn’t need to know how to combine them. The program focuses on stochastic simulation, but it is possible to extract the underlying deterministic matrix and some of its properties. However, in its standard form the model is rather inflexible: all individuals of a given sex reach reproductive maturity at the same age, and thereafter have unchanging demography. It is possible to build more complex life history models using functions, but this will be beyond the capacity of most non-modelers.
</dd>
</dl>
</div>
<div id="packages-for-parameter-estimation" class="section level3">
<h3>Packages for parameter estimation</h3>
<dl>
<dt>lmf: Functions for estimation and inference of selection in age-structured populations</dt>
<dd><strong>lmf</strong> <span class="citation">(Kvalnes 2013)</span> estimates directional and fluctuating selection in age-structured populations and provides methods for statistical inference using the procedures developed by Engen et al. 2012.
</dd>
<dt>poptrend: Analyze population trends from survey count data</dt>
<dd><strong>poptrend</strong> provides code to accompany <span class="citation">Knape (2016)</span>, who analyzed trends in Swedish bird populations using count data.
</dd>
</dl>
</div>
<div id="packages-for-human-population-analysis" class="section level3">
<h3>Packages for human population analysis</h3>
<dl>
<dt>bayesPop: Probabilistic Population Projection</dt>
<dd><strong>bayesPop</strong> <span class="citation">(Ševčı́ková and Raftery 2016)</span> produces stochastic projections of human populations in each country on Earth.
</dd>
<dt>demography: Forecasting mortality and fertility data</dt>
<dd>Functions for demographic analysis including lifetable calculations, Lee-Carter modelling and functional data analysis of mortality rates. <strong>demography</strong> <span class="citation">(Hyndman et al. 2017)</span> is focused on human population projections. Some of the life table calculations may be useful in ecological analysis.
</dd>
<dt>POPdemog: Plot Population Demographic History</dt>
<dd>Despite its intriguing title, <strong>POPdemog</strong> <span class="citation">(Zhou 2017)</span> actually provides visualizations of human population migration, based on coalescent analyses of genomes.
</dd>
<dt>LexisPlotR: Plot Lexis Diagrams for Demographic Purposes</dt>
<dd><strong>LexisPlotR</strong> <span class="citation">(Ottolinger 2016)</span> provides (not surprisingly) functions to create and display a <em>Lexis diagram</em>, a two dimensional diagram that is used to represent events (such as births or deaths) that occur to individuals belonging to different cohorts. It is employed in human demography; I’m not sure how useful it is in populations with discrete breeding, as all individuals in a cohort will be overlaid.
</dd>
</dl>
</div>
<div id="packages-for-other-specialized-populations" class="section level3">
<h3>Packages for other specialized populations</h3>
<dl>
<dt>albopictus: Age-Structured Population Dynamics Model</dt>
<dd><strong>albopictus</strong> <span class="citation">(Erguler 2018)</span> iterates a cohort of individuals forward with a daily timestep, incorporating mortality and development, and allowing new cohorts to be added extrinsically. It accompanies a pair of papers on mosquito population dynamics.
</dd>
<dt>capm: Companion Animal Population Management</dt>
<dd><strong>capm</strong> <span class="citation">(Santos Baquero, Amaku, and Ferreira 2015)</span> provides functions for creating and analyzing a population model of companion animals (e.g., domestic dogs and cats).
</dd>
<dt>pop.wolf: Models for Simulating Wolf Populations</dt>
<dd><strong>pop.wolf</strong> <span class="citation">(Chapron 2016)</span> is an R interface to a C program that runs an individual-based simulation of wolf population dynamics. There is little documentation; my guess is that this was code for a scientific paper (although no such paper is cited in the package).
</dd>
</dl>
</div>
</div>
<div id="roadmap" class="section level2">
<h2>Roadmap</h2>
<p>Version 0.1 is very basic: It contains only two construction functions (construct a Leslie matrix from a demography schedule; convert a prebreeding census model to a postbreeding census model) and one analysis tool (calculate the asymptotic growth rate). A great deal more needs to be done to make this useful.</p>
<div id="accomodate-the-two-commonest-use-cases-of-stage-structured-animal-mpms" class="section level3">
<h3>Accomodate the two commonest use cases of stage-structured animal MPMs</h3>
<p>Many animal populations modeled by stage-structured MPMs fall into one of two groups:</p>
<ol style="list-style-type: decimal">
<li><p>Stage-structured populations where all individuals in each stage have the same demography, and the maturation from one stage to the next occurs at a fixed age. This is commonly used for long-lived mammals and reptilians [e.g., sea turtles; <span class="citation">Crowder et al. (1994)</span>]. We call these “stage-for-age” models.</p></li>
<li><p>Size-structured populations in which growth is approximately deterministic (e.g., via a von Bertalanffy function) and demography is size-dependent, and the analyst wants to collapse the size distribution into a small number of size/stage classes. These are commonly used for long-lived fish [e.g., XXX]. We call these “size-for-age” models; conceptually, they differ from stage-for-age models by having substantial demographic heterogeneity within each size class.</p></li>
</ol>
<p>There are two general approaches for dealing with such populations:</p>
<ol style="list-style-type: decimal">
<li><p>Construct stage-structured MPMs that approximate as closely as possible the full age-structured MPM <span class="citation">(Crowder et al. 1994; Caswell 2001; Fujiwara and Diaz-Lopez 2017)</span>.</p></li>
<li><p>Construct the full age-structured MPM, and then collapse the results of the analyis back into the smaller number of stages or size classes.</p></li>
</ol>
<p>Building functions to do this is the next priority of the project. The second part will require creating the first iteration of an “mpm-class” that tracks stage/size class along with age. It will also require some research on how to aggregate the reproductive value, as I don’t think we got it right in our paper.</p>
</div>
<div id="accomodate-multi-sex-models" class="section level3">
<h3>Accomodate multi-sex models</h3>
<p>Functions to create and manipulate models that explicitly track males and females. This includes a frequency-dependent mating function. Also allow newborns to transition into multiple classes. This is also the place to add clonal reproduction and fission.</p>
</div>
<div id="use-periodic-models-to-accomodate-more-flexible-descriptions-of-life-history" class="section level3">
<h3>Use periodic models to accomodate more flexible descriptions of life history</h3>
<p>In many cases, it may be desireable to incorporate seasonal variation in demography as well as to separate mating from birth. This can be done by letting the user specify individual transition events (including some, such as births, that don’t have a duration), and using the theory of periodic models to assemble them into annual models. While the end result is a single matrix, indicating that, in principle, the matrix elements could be calculated directly, this approach will probably be easier for the user to apply accurately.</p>
<p>Furthermore, this is probably the way to flexibly incorporate analysis in terms of underlying parameters, as even a standard model can be written as a product of matrices reflecting births, survival, and stage transitions. At this point we will include a full set of asymptotic analysis functions.</p>
</div>
<div id="provide-a-link-to-ipms" class="section level3">
<h3>Provide a link to IPMs</h3>
<p>The goal here is to allow the analysis of integral projection models (e.g., from <strong>IPMpack</strong>) to be summarized by size classes. Also consider other ways in which continuous state variables might be included (i.e., creating our own IPM constructor function).</p>
</div>
<div id="stochastic-mpms" class="section level3">
<h3>Stochastic MPMs</h3>
<p>Extend the mpm-class to incorporate the information needed to represent environmental stochasticity (multiple observed matrices, temporal parameter covariances) and demographic stochasticity (distributional forms, variance parameters). Also write functions to project and analyze these models (e.g., sensitivity analysis of stochastic growth rate).</p>
</div>
<div id="density-dependence" class="section level3">
<h3>Density dependence</h3>
<p>I don’t yet have a clear view for what to do with density dependence. But it seems likely that many empirical modelers would incorporate this if they could. The periodic breakdown of population processes may be particularly powerful here.</p>
</div>
<div id="other-analyses" class="section level3">
<h3>Other analyses</h3>
<p>Catalog and prioritize the nonstandard analyses from Caswell and elsewhere. Investigate the vec-permutation formalism for compactly representing more complex models.</p>
</div>
<div id="parameter-estimation" class="section level3">
<h3>Parameter estimation</h3>
<p>For now, parameter estimation is not a high priority to me, other than correctly estimating stage transition rates. It appears that estimation of (st)age-specific demographic rates is not a limiting step (although I may be wrong). However, there are some deep conceptual issues around estimating the mean and variance of maturation times when individuals don’t all have identical stage durations. In addition, there would be value in developing ways to bring standard errors and covariances of parameter estimates into the models, in order to propogate those uncertainties to the analytical outputs.</p>
</div>
</div>
<div id="references" class="section level2 unnumbered">
<h2 class="unnumbered">References</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Caswell2001" class="csl-entry">
Caswell, H. 2001. <em>Matrix Population Models: Construction, Analysis, and Interpretation</em>. 2nd ed. Sunderland, MA: Sinauer Associates.
</div>
<div id="ref-Chapron2016" class="csl-entry">
Chapron, Guillaume. 2016. <em>Pop.wolf: Models for Simulating Wolf Populations</em>. <a href="https://CRAN.R-project.org/package=pop.wolf">https://CRAN.R-project.org/package=pop.wolf</a>.
</div>
<div id="ref-Chapron2018r" class="csl-entry">
———. 2018. <em>Population: Models for Simulating Populations</em>. <a href="https://CRAN.R-project.org/package=population">https://CRAN.R-project.org/package=population</a>.
</div>
<div id="ref-Crowder1994" class="csl-entry">
Crowder, L. B., D. T. Crouse, S. S. Heppell, and T. H. Martin. 1994. <span>“<span class="nocase">Predicting the impact of turtle excluder devices on loggerhead sea turtle populations</span>.”</span> <em>Ecol Appl</em> 4: 437–45. <a href="https://doi.org/10.2307/1941948">https://doi.org/10.2307/1941948</a>.
</div>
<div id="ref-delaCruzRot2017r" class="csl-entry">
de la Cruz Rot, Marcelino. 2017. <em><span>Rramas</span>: Matrix Population Models</em>. <a href="https://CRAN.R-project.org/package=Rramas">https://CRAN.R-project.org/package=Rramas</a>.
</div>
<div id="ref-Ellner2016a" class="csl-entry">
Ellner, S. P., D. Z. Childs, and M. Rees. 2016. <em>Data-Driven Modelling of Structured Populations: A Practical Guide to the Integral Projection Model</em>. Switzerland: Springer International. <a href="http://www.amazon.com/Data-driven-Modelling-Structured-Populations-Mathematical/dp/3319288911%3FSubscriptionId%3D0NM5T5X751JWT17C4GG2%26tag%3Dsonnysoftware%26linkCode%3Dxm2%26camp%3D2025%26creative%3D165953%26creativeASIN%3D3319288911">http://www.amazon.com/Data-driven-Modelling-Structured-Populations-Mathematical/dp/3319288911%3FSubscriptionId%3D0NM5T5X751JWT17C4GG2%26tag%3Dsonnysoftware%26linkCode%3Dxm2%26camp%3D2025%26creative%3D165953%26creativeASIN%3D3319288911</a>.
</div>
<div id="ref-Erguler2018r" class="csl-entry">
Erguler, Kamil. 2018. <em>Albopictus: Age-Structured Population Dynamics Model</em>. <a href="https://CRAN.R-project.org/package=albopictus">https://CRAN.R-project.org/package=albopictus</a>.
</div>
<div id="ref-Fujiwara2017" class="csl-entry">
Fujiwara, M., and J. Diaz-Lopez. 2017. <span>“<span class="nocase">Constructing stage-structured matrix population models from life tables: comparison of methods.</span>”</span> <em>PeerJ</em> 5: e3971. <a href="https://doi.org/10.7717/peerj.3971">https://doi.org/10.7717/peerj.3971</a>.
</div>
<div id="ref-Golding2016" class="csl-entry">
Golding, Nick. 2016. <em>Pop: A Flexible Syntax for Population Dynamic Modelling</em>. <a href="https://CRAN.R-project.org/package=pop">https://CRAN.R-project.org/package=pop</a>.
</div>
<div id="ref-Hyndman2017" class="csl-entry">
Hyndman, Rob J, Heather Booth, Leonie Tickle, and John Maindonald. 2017. <em>Demography: Forecasting Mortality, Fertility, Migration and Population Data</em>. <a href="https://CRAN.R-project.org/package=demography">https://CRAN.R-project.org/package=demography</a>.
</div>
<div id="ref-Kendall_PVA2018r" class="csl-entry">
Kendall, Bruce E. 2018. <em>PVA: Population Viability Analysis in r</em>. <a href="https://github.com/BruceKendall/PVA">https://github.com/BruceKendall/PVA</a>.
</div>
<div id="ref-Kendall2019ir" class="csl-entry">
Kendall, Bruce E., Masami Fujiwara, Jasmin Diaz-Lopez, Sandra Schneider, Jakob Voigt, and Sören Wiesner. 2019. <span>“Persistent Problems in the Construction of Matrix Population Models.”</span> <em>Ecological Modelling</em> in review.
</div>
<div id="ref-Knape2016" class="csl-entry">
Knape, Jonas. 2016. <span>“Decomposing Trends in Swedish Bird Populations Using Generalized Additive Mixed Models.”</span> <em>Journal of Applied Ecology</em> 53: 1852–61. <a href="https://doi.org/10.1111/1365-2664.12720">https://doi.org/10.1111/1365-2664.12720</a>.
</div>
<div id="ref-Kvalnes2013" class="csl-entry">
Kvalnes, Thomas. 2013. <em>Lmf: Functions for Estimation and Inference of Selection in Age-Structured Populations</em>. <a href="https://CRAN.R-project.org/package=lmf">https://CRAN.R-project.org/package=lmf</a>.
</div>
<div id="ref-Metcalf2014r" class="csl-entry">
Metcalf, CJE, SM McMahon, R Salguero-Gomez, E Jongejans, and C Merow. 2014. <em>IPMpack: Builds and Analyses Integral Projection Models (IPMs).</em> <a href="https://CRAN.R-project.org/package=IPMpack">https://CRAN.R-project.org/package=IPMpack</a>.
</div>
<div id="ref-Morris2002a" class="csl-entry">
Morris, W. F., and D. F. Doak. 2002. <em>Quantitative Conservation Biology: Theory and Practice of Population Viability Analysis</em>. Sunderland, MA: Sinauer Associates. <a href="http://books.google.com/books?id=8Z6GQgAACAAJ&amp;printsec=frontcover">http://books.google.com/books?id=8Z6GQgAACAAJ&amp;printsec=frontcover</a>.
</div>
<div id="ref-Ottolinger2016r" class="csl-entry">
Ottolinger, Philipp. 2016. <em>LexisPlotR: Plot Lexis Diagrams for Demographic Purposes</em>. <a href="https://CRAN.R-project.org/package=LexisPlotR">https://CRAN.R-project.org/package=LexisPlotR</a>.
</div>
<div id="ref-SantosBaquero2015" class="csl-entry">
Santos Baquero, Oswaldo, Marcos Amaku, and Fernando Ferreira. 2015. <span>“Capm: Companion Animal Population Management. R Package Version 0.9.0.”</span> <a href="http://oswaldosantos.github.io/capm">http://oswaldosantos.github.io/capm</a>.
</div>
<div id="ref-Stevens2009" class="csl-entry">
Stevens, M. H. H. 2009. <em>A Primer of Ecology with r, 2nd Printing</em>. New York: Springer.
</div>
<div id="ref-Stott2018r" class="csl-entry">
Stott, Iain, Dave Hodgson, and Stuart Townley. 2018. <em>Popdemo: Demographic Modelling Using Projection Matrices</em>. <a href="https://CRAN.R-project.org/package=popdemo">https://CRAN.R-project.org/package=popdemo</a>.
</div>
<div id="ref-Stubben2007" class="csl-entry">
Stubben, C., and B. Milligan. 2007. <span>“<span class="nocase">Estimating and analyzing demographic models using the popbio package in R</span>.”</span> <em>J Stat Soft</em> 22: 11. <a href="https://doi.org/10.18637/jss.v022.i11">https://doi.org/10.18637/jss.v022.i11</a>.
</div>
<div id="ref-Sevcikova2016" class="csl-entry">
Ševčı́ková, Hana, and Adrian E. Raftery. 2016. <span>“<span class="nocase">bayesPop</span>: Probabilistic Population Projections.”</span> <em>Journal of Statistical Software</em> 75 (5): 1–29. <a href="https://doi.org/10.18637/jss.v075.i05">https://doi.org/10.18637/jss.v075.i05</a>.
</div>
<div id="ref-Zhou2017" class="csl-entry">
Zhou, Ying. 2017. <em>POPdemog: Plot Population Demographic History</em>. <a href="https://CRAN.R-project.org/package=POPdemog">https://CRAN.R-project.org/package=POPdemog</a>.
</div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
